Chapters/Common/StringAlgorithms.cs

```diff
@@ -0,0 +1,315 @@
+
+            for (int i = 0; i < Input.Length; i += step)
+            {
+
+                if (char.IsDigit(Input[i]) && !isFirst)
+                {
+                    isFirst = true;
+                    start = i;
+                }
+                else
+                {
+                    isFirst = false;
+                    step = i - start;
+
+                    byte[] num = new byte[step];
+                    for (int j = start; j < i; j++)
+                        num[i - j] = Convert.ToByte(Input[j]);
+                    for (int j = num.Length - 1; j > 0; j--)
+                        number += num[j] * (int)Math.Pow(10, num.Length - 1 - j);
+                    if (BinaryFile.IsSimple(number))
+                        sum += number;
+                }
+            }
+            return sum;
+        }
+
+
+        //10
+        public void RemoveMaxLengthWord()
+        {
+            int temp = 0;
+            int start = 0;
+            bool isFirst = false;
+            int[] word = new [] { 0, 0};
+
+            //Find 
+            for (int i = 0; i < Input.Length; i++)
+            {
+                if (char.IsLetter(Input[i]) && !isFirst)
+                {
+                    isFirst = true;
+                    start = i;
+                    temp++;
+                }
+                if (char.IsLetter(Input[i]) && isFirst)
+                    temp++;
+                else
+                {
+                    if (temp > word[1])
+                    {
+                        word[0] = start;
+                        word[1] = temp;
+                    }
+                    isFirst = false;
+                    temp = 0;
+                    start = 0;
+                }
+            }
+
+            Input.Remove(word[0], word[1]);
+        }
+
+        //11
+        public int FindSimpleNumbers()
+        {
+            int step = 1;
+            int count = 0;
+            int start = 0;
+            int number = 0;
+            bool isFirst = false;
+
+            for (int i = 0; i < Input.Length; i += step)
+            {
+                if (char.IsDigit(Input[i]) && !isFirst)
+                {
+                    isFirst = true;
+                    start = i;
+                }
+                else
+                {
+                    isFirst = false;
+                    step = i - start;
+
+                    byte[] num = new byte[step];
+                    for (int j = start; j < i; j++)
+                        num[i - j] = Convert.ToByte(Input[j]);
+                    for (int j = num.Length - 1; j > 0; j--)
+                        number += num[j] * (int)Math.Pow(10, num.Length - 1 - j);
+                    if (BinaryFile.IsSimple(number))
+                        count++;
+                }
+            }
+            return count;
+        }
+
+        //12
+        public void InsertSpaceAfterPunctuation()
+        {
+            for (int i = 0; i < Input.Length; i++)
```

Comment:
This is not a bug, but try to optimize this. There should be redundant iterations, and this is a bicycle.